include/nuttx/sched.h

```diff
@@ -525,6 +526,9 @@ struct tcb_s
 
   /* Task Group *************************************************************/
 
+#ifdef CONFIG_ARCH_ADDRENV
+  FAR struct task_group_s *mm_group;   /* Active memory mappings group      */
```

Comment:
why need mm_group and group?

Reply:
Another option would be to store the address environment only here but that gets a bit dirty IMO. It would require changing group->addrenv into a pointer and allocating the addrenv from kheap. 

My reason for not doing this is that each group needs the address environment, so why try allocating it separately from heap ?

Reply:
I think one of my comments got lost, the reason we need mm_group and group separately, is that when a process is destroyed it detaches the group from tcb->group in group_leave.c 
```

void group_leave(FAR struct tcb_s *tcb)
{
...
      /* In any event, we can detach the group from the TCB so that we won't
       * do this again.
       */
      tcb->group = NULL;
    }
}
```

This code is good and I would not modify it.